BinaryEncoding.md

```diff
@@ -63,17 +57,14 @@ sequence, then followed recursively by any child nodes.
 
 * Examples
   * Given a simple AST node: `I32Add(left: AstNode, right: AstNode)`
-    * First write the opcode for `I32Add` (uint8)
+    * First write the opcode for `I32Add` (`varuint32`)
```

Comment:
For opcodes there could be more special tailored schemes, with secondary opcodes just in selected cases. For example, for SIMD in particular, you could imagine that there is just one primary opcode SIMD, and the zoo of actual operations is distinguished by a secondary byte. (Or alternatively, there could be one primary opcode per SIMD type.)

I think the advantage is that this provides a larger short opcode space (up to almost 2x), and a bit more flexibility in reserving/prioritising short opcodes for selected future operators -- opcode table notwithstanding.